app/main.py

```diff
@@ -1,3 +1,37 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: float) -> object:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: float) -> object:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: float) -> object:
+        return Distance(round(self.km / other, 2))
+
```

Comment:
in __mul__ and __truediv__ methods other can be int or float, in the rest of the methods int, float or an  instance of the Distance class. 

Add logic to handle cases when other is an  instance of the Distance class and fix type hints.

Reply:
I avoided unnecessary logic, as I was advised last time, there were 110 questions in the test, which my solution passed successfully, how can I understand where the logic is excessive and where it is insufficient?